### ISSUE TITLE
2.2 regresson: junos_configure: self.raw_decode(s, idx=_w(s, 0).end())\nTypeError: expected string or buffer\n", 

### ISSUE THREAD

**Issue by gundalow** [ansible/ansible#20186]
##### COMPONENT NAME
junos_configure

##### ANSIBLE VERSION
```
ansible 2.2.1.0 (detached HEAD fe33c937c4) last updated 2016/12/15 12:03:42 (GMT +100)
  lib/ansible/modules/core: (detached HEAD 48a3ce1223) last updated 2016/12/15 12:04:11 (GMT +100)
  lib/ansible/modules/extras: (detached HEAD 4e720b48ae) last updated 2016/12/15 12:04:18 (GMT +100)
```

##### STEPS TO REPRODUCE
2.2 regression. Fails on `vsrx01.eng` and AWS. Can't test on `devel` as junos hasn't been refactored yet.

ansible/test/integration/targets/junos_config/tests/netconf/src_basic.yaml
```yaml
---
- debug: msg="START netconf/src_basic.yaml"

- name: setup
  junos_config:
    lines:
      - set system host-name {{ inventory_hostname_short }}
      - delete interfaces lo0
    provider: "{{ netconf }}"

- name: configure device with config
  junos_config:
    src: basic/config.j2
    provider: "{{ netconf }}"
  register: result

- assert:
    that:
      - "result.changed == true"
# https://github.com/ansible/ansible-modules-core/issues/4807
      - "result.updates is not defined"

```

##### ACTUAL RESULTS
```
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_BOpQw7/ansible_module_junos_config.py\", line 343, in <module>\n    main()\n  File \"/tmp/ansible_BOpQw7/ansible_module_junos_config.py\", line 334, in main\n    run(module, result)\n  File \"/tmp/ansible_BOpQw7/ansible_module_junos_config.py\", line 292, in run\n    return load_config(module, result)\n  File \"/tmp/ansible_BOpQw7/ansible_module_junos_config.py\", line 245, in load_config\n    config_format = module.params['src_format'] or guess_format(candidate)\n  File \"/tmp/ansible_BOpQw7/ansible_module_junos_config.py\", line 176, in guess_format\n    json.loads(config)\n  File \"/usr/lib/python2.7/json/__init__.py\", line 339, in loads\n    return _default_decoder.decode(s)\n  File \"/usr/lib/python2.7/json/decoder.py\", line 364, in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\nTypeError: expected string or buffer\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE"
}
```

**Comment by gundalow** [comment]
@Qalthos Junos issue for you

**Comment by abadger** [comment]
Not sure if you'll need this here, but just in case: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L340  there's a pair of functions to recursively walk through a dictionary/list and turn everything into unicode or bytes.  Note that it can be expensive to recurse over a large amount of data so if you know that it's just a few values which are giving you problems, it can be better to just touch the few.

**Comment by Qalthos** [comment]
As I expected, once the dust settled on 2.3 modules for Junos, this issue disappeared. I am considering this issue to be solved for 2.3
